### PR TITLE
FEATURE Variation Inventory Manager

### LINKED ISSUE #24: FEATURE Variation Inventory management
**Is your feature request related to a problem? Please describe.**
add the ability to track inventory on product Variations, set to unavailable once inventory is depleted

### PR DESCRIPTION
**Pull request by jsirish**
closes #24

### CODE REVIEW

File: src/Extension/ProductInventoryManager.php [hunk 1]
```diff
@@ -104,7 +112,21 @@ public function getNumberPurchased()
     public function getOrders()
     {
         if ($this->owner->ID) {
-            return OrderDetail::get()->filter('ProductID', $this->owner->ID);
```
Comment (jsirish) [inline src/Extension/ProductInventoryManager.php:107]: Seems like an opportunity to make this more efficient

File: src/Extension/ProductVariationInventoryManager.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+use Dynamic\Foxy\Orders\Model\OrderVariation;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataList;
+
+class ProductVariationInventoryManager extends ProductInventoryManager
+{
+    /**
+     * @return DataList
+     */
+    public function getOrders()
```
Comment (jsirish) [inline src/Extension/ProductVariationInventoryManager.php:15]: Seems like an opportunity to make this more efficient

File: src/Extension/ProductInventoryManager.php [hunk 3]
```diff
@@ -104,7 +112,21 @@ public function getNumberPurchased()
     public function getOrders()
```
Comment (jsirish) [inline src/Extension/ProductInventoryManager.php:112]: Seems like an opportunity to make this more efficient

**Comment by jsirish** [comment]
Is this likely a 1.1.0 release?

File: src/Extension/ProductInventoryManager.php (hunk 3 above)
Comment (muskie9) [reply src/Extension/ProductInventoryManager.php:112]: I think there are some efficiencies that could be gained, but that can be a good next release ticket.

File: src/Extension/ProductVariationInventoryManager.php (hunk 2 above)
Comment (muskie9) [reply src/Extension/ProductVariationInventoryManager.php:15]: Same as other inventory manager class.

**Review by muskie9 — APPROVED** [review]
I think this looks solid overall!

File: src/Extension/ProductInventoryManager.php (hunk 3 above)
Comment (jsirish) [reply src/Extension/ProductInventoryManager.php:112]: 👍

**Pull request merged**